### PR TITLE
DOM logic cleanup; fix continue button race condition; fix output wrapping

### PR DESCRIPTION
**Pull request by chenglou**
Some more nits. Since many of the dom elements are lifted up into references, I did so for the rest too

The continue button wasn't disabled, so there were race conditions when you click it during generation. This PR disables it

The output didn't wrap around:
image

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -57,82 +62,79 @@ <h1>WebGPU GPT Model Demo</h1>
       }
 
       async function startGeneration() {
-        setTextareaVisibility(true);
```
Comment (chenglou) [inline index.html:60]: so "visibility true" was a little confusing. It's really setting "visibility" to false

File: index.html [hunk 2]
```diff
@@ -57,82 +62,79 @@ <h1>WebGPU GPT Model Demo</h1>
...
-        document.getElementById("temperatureInput").value = 1;
+        temperatureInput.value = 1;
       }
 
-      function setTextareaVisibility(bool) {
-        const promptTextarea = document.getElementById("prompt");
+      function setTextareaDisabled(bool) {
         promptTextarea.disabled = bool;
         generateButton.disabled = bool;
+        continueButton.disabled = bool;
```
Comment (chenglou) [inline index.html:118]: This is the missing continue button disabling line

File: index.html [hunk 3]
```diff
@@ -28,22 +28,27 @@ <h1>WebGPU GPT Model Demo</h1>
...
+WILL:
 Ah, how dare you challenge me?
 Have you forgotten I built WebGPT? </textarea
     >
     <br />
     <button id="generateButton" onclick="startGeneration()" disabled>Generate Text</button>
     <button id="continueButton" onclick="continueGeneration()" disabled>Continue Generation</button>
     <br /><br />
-    <span id="output" style="white-space: pre"></span>
+    <p id="output" style="white-space: pre-wrap"></p>
```
Comment (chenglou) [inline index.html:39]: Output wrapping

**Comment by chenglou** [comment]
In fact, why not just output directly into the text area and forget about the continue button?

**Comment by 0hq** [comment]
Thanks, totally makes sense.

**Pull request merged**